tls/README.md

```diff
@@ -0,0 +1,382 @@
+* implement `Requirement` interface from `pkg/cmd/update-tls-artifacts/generate-owners/tlsmetadata/tlsmetadatainterfaces/types.:go`:
+```golang
+type Requirement interface {
+	GetName() string
+
+	// InspectRequirement generates and returns the result for a particular set of raw data
+	InspectRequirement(rawData []*certgraphapi.PKIList) (RequirementResult, error)
+}
+```
+
+Example:
+```golang
+type OwnerRequirement struct {
+	name string
+}
+
+func NewOwnerRequirement() tlsmetadatainterfaces.Requirement {
+	return OwnerRequirement{
+		name: "ownership",
+	}
+}
+
+// Return requirement name for filenames. Must by unique
+func (o OwnerRequirement) GetName() string {
+	return o.name
+}
+
+// Boilerplate function which processes TLS registry and produces RequirementResult or throws error
+func (o OwnerRequirement) InspectRequirement(rawData []*certgraphapi.PKIList) (tlsmetadatainterfaces.RequirementResult, error) {
+	pkiInfo, err := tlsmetadatainterfaces.ProcessByLocation(rawData)
+	if err != nil {
+		return nil, fmt.Errorf("transforming raw data %v: %w", o.GetName(), err)
+	}
+
+	ownershipJSONBytes, err := json.MarshalIndent(pkiInfo, "", "    ")
+	if err != nil {
+		return nil, fmt.Errorf("failure marshalling %v.json: %w", o.GetName(), err)
+	}
+	markdown, err := generateOwnershipMarkdown(pkiInfo)
+	if err != nil {
+		return nil, fmt.Errorf("failure marshalling %v.md: %w", o.GetName(), err)
+	}
+	violations := generateViolationJSON(pkiInfo)
+	violationJSONBytes, err := json.MarshalIndent(violations, "", "    ")
+	if err != nil {
+		return nil, fmt.Errorf("failure marshalling %v-violations.json: %w", o.GetName(), err)
+	}
+
+	return tlsmetadata.NewRequirementResult(
+		o.GetName(),
+		ownershipJSONBytes,
+		markdown,
+		violationJSONBytes)
+}
+```
+
+This interface calls two internal functions:
+```golang
+func generateViolationJSON(pkiInfo *certgraphapi.PKIRegistryInfo) *certgraphapi.PKIRegistryInfo {
+	ret := &certgraphapi.PKIRegistryInfo{}
+
+  // Iterate over certificate-key pairs (==secrets)
+	for i := range pkiInfo.CertKeyPairs {
+		curr := pkiInfo.CertKeyPairs[i]
+		owner := curr.CertKeyInfo.OwningJiraComponent
+    // Include in violations list if has no owner set or owner set to Unknown
+		if len(owner) == 0 || owner == tlsmetadata.UnknownOwner {
+			ret.CertKeyPairs = append(ret.CertKeyPairs, curr)
+		}
+	}
+
+  // Iterate over CA bundles (==configmaps)
+	for i := range pkiInfo.CertificateAuthorityBundles {
+		curr := pkiInfo.CertificateAuthorityBundles[i]
+		owner := curr.CABundleInfo.OwningJiraComponent
+    // Include in violations list if has no owner set or owner set to Unknown
+		if len(owner) == 0 || owner == tlsmetadata.UnknownOwner {
+			ret.CertificateAuthorityBundles = append(ret.CertificateAuthorityBundles, curr)
+		}
+	}
+
+	return ret
+}
+```
+
+Markdown report can also be customized:
+```golang
+func generateOwnershipMarkdown(pkiInfo *certgraphapi.PKIRegistryInfo) ([]byte, error) {
```

Comment:
indicate method and provide link.  I think once we generalize the annotation based requirements most people won't need this.